pkg/services/provisioning/datasources/types.go

```diff
@@ -135,6 +137,12 @@ func (cfg *DatasourcesAsConfigV1) mapToDatasourceFromConfig(apiVersion int64) *D
 			Editable:          ds.Editable,
 			Version:           ds.Version,
 		})
+		if ds.Password != "" {
```

Comment:
Added warnings here, thought not sure if there won't be too much of them if somebody has a lots of datasources.